restapi.go

```diff
@@ -1785,13 +1786,13 @@ func (s *Session) ChannelInviteCreate(channelID string, i Invite) (st *Invite, e
 // ChannelPermissionSet creates a Permission Override for the given channel.
 // NOTE: This func name may changed.  Using Set instead of Create because
 // you can both create a new override or update an override with this function.
-func (s *Session) ChannelPermissionSet(channelID, targetID, targetType string, allow, deny int) (err error) {
+func (s *Session) ChannelPermissionSet(channelID, targetID string, targetType PermissionType, allow, deny int) (err error) {
 
 	data := struct {
-		ID    string `json:"id"`
-		Type  string `json:"type"`
-		Allow int    `json:"allow"`
-		Deny  int    `json:"deny"`
+		ID    string         `json:"id"`
+		Type  PermissionType `json:"type"`
+		Allow int            `json:"allow"`
+		Deny  int            `json:"deny"`
```

Comment:
These should probably also be marked as `,string`. They seem to work currently, but I'm guessing that we're hitting some non-guaranteed coercion behavior on the API's side, which could break if they add more permissions.